src/apps/stable/features/playback/utils/mediaSessionSubscriber.ts

```diff
@@ -0,0 +1,178 @@
+        if (hasNavigatorSession) this.bindNavigatorSession();
+    }
+
+    private bindNavigatorSession() {
+        /* eslint-disable compat/compat */
+        navigator.mediaSession.setActionHandler('pause', this.onMediaSessionAction.bind(this));
+        navigator.mediaSession.setActionHandler('play', this.onMediaSessionAction.bind(this));
+        navigator.mediaSession.setActionHandler('stop', this.onMediaSessionAction.bind(this));
+        navigator.mediaSession.setActionHandler('previoustrack', this.onMediaSessionAction.bind(this));
+        navigator.mediaSession.setActionHandler('nexttrack', this.onMediaSessionAction.bind(this));
+        navigator.mediaSession.setActionHandler('seekto', this.onMediaSessionAction.bind(this));
+        // iOS will only show next/prev track controls or seek controls
+        if (!browser.iOS) {
+            navigator.mediaSession.setActionHandler('seekbackward', this.onMediaSessionAction.bind(this));
+            navigator.mediaSession.setActionHandler('seekforward', this.onMediaSessionAction.bind(this));
+        }
+        /* eslint-enable compat/compat */
+    }
+
+    private onMediaSessionAction(details: MediaSessionActionDetails) {
+        switch (details.action) {
+            case 'pause':
+                return this.playbackManager.pause(this.player);
+            case 'play':
+                return this.playbackManager.unpause(this.player);
+            case 'stop':
+                return this.playbackManager.stop(this.player);
+            case 'seekbackward':
+                return this.playbackManager.rewind(this.player);
+            case 'seekforward':
+                return this.playbackManager.fastForward(this.player);
+            case 'seekto':
+                return this.playbackManager.seekMs((details.seekTime || 0) * MILLISECONDS_PER_SECOND, this.player);
+            case 'previoustrack':
+                return this.playbackManager.previousTrack(this.player);
+            case 'nexttrack':
+                return this.playbackManager.nextTrack(this.player);
+            default:
+                console.info('[MediaSessionSubscriber] Unhandled media session action', details);
+        }
+    }
+
+    private onMediaSessionUpdate(
+        { type: action }: Event,
+        state: PlayerState = this.playbackManager.getPlayerState(this.player)
+    ) {
+        const item = state.NowPlayingItem;
+
+        if (!item) {
+            console.debug('[MediaSessionSubscriber] no now playing item; resetting media session', state);
+            return resetMediaSession();
+        }
+
+        const isVideo = item.MediaType === MediaType.Video;
+        const isLocalPlayer = !!this.player?.isLocalPlayer;
+
+        // Local players do their own notifications
+        if (isLocalPlayer && isVideo) {
+            console.debug('[MediaSessionSubscriber] ignoring local player update');
+            return;
+        }
+
+        const album = item.Album || undefined;
```

Comment:
Better to use ??

At some point we should tackle the annoying `null` introduced by the SDK 😅.

Reply:
Dmitry doesn't like `??` because the code babel produces for it is really verbose.

Agreed it would be really nice if the SDK only used `undefined` though. :upside_down_face: